Makefile

```diff
@@ -1,21 +1,77 @@
+SHELL := bash
+.ONESHELL:
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
+MAKEFLAGS += --warn-undefined-variables
+MAKEFLAGS += --no-builtin-rules
+MAKEFLAGS += --no-print-directory
+MAKEFLAGS += -S
+
 .DEFAULT: all
 
+CARGO ?= cargo
+CARGO_WATCH ?= cargo-watch
+CARGO_CLIPPY ?= cargo-clippy
+DOCKER ?= docker
+
 all: build test
 
-build:
-	cargo build --features cli
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_\-.*]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+###########
+# Tooling #
+###########
+
+# Ensure that cargo watch is installed
+check-cargo-watch:
+ifeq ("",$(shell command -v $(CARGO_WATCH)))
+	$(error "ERROR: cargo-watch is not installed (see: https://crates.io/crates/cargo-watch)")
+endif
+
+# Ensure that clippy is installed
+check-cargo-clippy:
+ifeq ("",$(shell command -v $(CARGO_CLIPPY)))
+	$(error "ERROR: clippy is not installed (see: https://doc.rust-lang.org/clippy/installation.html)")
+endif
+
+#########
+# Build #
+#########
 
+lint: check-cargo-clippy
+	$(CARGO) fmt --all --check
+	$(CARGO) clippy --all-features --all-targets --workspace
+
+build: ## Build wadm
+	$(CARGO) build
+
+build-watch: check-cargo-watch ## Build wadm continuously
+	$(CARGO) watch -- $(MAKE) build

```

Comment:
Is a Makefile directive really required for this?

Reply:
It's not required, but the important bit is to have a place to helpfully tell the user what is wrong and *how to fix it*. `check-cargo-watch` notes what/where `cargo-watch` is so that contributors can fix a little easier.